src/Microsoft.Extensions.DependencyInjection/ServiceProvider.cs

```diff
@@ -186,7 +186,7 @@ public void Dispose()
             }
         }
 
-        internal object CaptureDisposable(object service)
+        internal virtual object CaptureDisposable(object service)
```

Comment:
That's still worth doing as this code path gets used for singletons. Another PR 😄